br/pkg/lightning/backend/external/kv_reader.go

```diff
@@ -40,7 +40,7 @@ func newKVReader(
 	if err != nil {
 		return nil, err
 	}
-	br, err := newByteReader(ctx, sr, bufSize)
+	br, err := newByteReader(ctx, sr, bufSize, store, name, false)
```

Comment:
considering `newByteReader` will re-open the storage reader when switching, now I think we can also move the logic of opening `sr` from `newKVReader` to `newByteReader`. The fucntion parameters will be less.